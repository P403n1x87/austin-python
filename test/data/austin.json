{
  "$schema": "https://www.speedscope.app/file-format-schema.json",
  "shared": {
    "frames": [
      {
        "name": "_bootstrap",
        "file": "/usr/lib/python3.8/threading.py",
        "line": 890
      },
      {
        "name": "_bootstrap_inner",
        "file": "/usr/lib/python3.8/threading.py",
        "line": 932
      },
      {
        "name": "run",
        "file": "/usr/lib/python3.8/threading.py",
        "line": 870
      },
      {
        "name": "keep_cpu_busy",
        "file": "/home/gabriele/Projects/austin/test/target34.py",
        "line": 32
      },
      {
        "name": "<module>",
        "file": "/home/gabriele/Projects/austin/test/target34.py",
        "line": 37
      },
      {
        "name": "start",
        "file": "/usr/lib/python3.8/threading.py",
        "line": 857
      },
      {
        "name": "wait",
        "file": "/usr/lib/python3.8/threading.py",
        "line": 558
      },
      {
        "name": "wait",
        "file": "/usr/lib/python3.8/threading.py",
        "line": 302
      },
      {
        "name": "<module>",
        "file": "/home/gabriele/Projects/austin/test/target34.py",
        "line": 38
      },
      {
        "name": "keep_cpu_busy",
        "file": "/home/gabriele/Projects/austin/test/target34.py",
        "line": 31
      },
      {
        "name": "keep_cpu_busy",
        "file": "/home/gabriele/Projects/austin/test/target34.py",
        "line": 30
      }
    ]
  },
  "profiles": [
    {
      "name": "Wall time profile for 82848:82848",
      "unit": "microseconds",
      "startValue": 0,
      "endValue": 372856,
      "samples": [
        [
          4,
          5,
          6,
          7
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          10
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          10
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          10
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          9
        ],
        [
          8,
          9
        ],
        [
          8,
          9
        ],
        [
          8,
          9
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8,
          3
        ],
        [
          8
        ]
      ],
      "weights": [
        10058,
        10083,
        10088,
        10088,
        10087,
        10060,
        10085,
        10082,
        10020,
        10090,
        10057,
        10058,
        10087,
        10087,
        10083,
        10060,
        10057,
        10083,
        10094,
        10058,
        10058,
        10057,
        10081,
        10087,
        10087,
        10087,
        10081,
        10088,
        10084,
        10089,
        10089,
        10084,
        10088,
        10090,
        10091,
        10058,
        10092
      ],
      "type": "sampled"
    },
    {
      "name": "Wall time profile for 82848:82858",
      "unit": "microseconds",
      "startValue": 0,
      "endValue": 352706,
      "samples": [
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          10
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          9
        ],
        [
          0,
          1,
          2,
          10
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2,
          3
        ],
        [
          0,
          1,
          2
        ],
        [
          0,
          1,
          2
        ]
      ],
      "weights": [
        10058,
        10083,
        10088,
        10088,
        10087,
        10060,
        10085,
        10082,
        10020,
        10090,
        10057,
        10058,
        10087,
        10087,
        10083,
        10060,
        10057,
        10083,
        10094,
        10058,
        10058,
        10057,
        10081,
        10087,
        10087,
        10087,
        10081,
        10088,
        10084,
        10089,
        10089,
        10084,
        10088,
        10090,
        10091
      ],
      "type": "sampled"
    }
  ],
  "name": "austin.out",
  "exporter": "Austin2Speedscope Converter 0.2.0"
}