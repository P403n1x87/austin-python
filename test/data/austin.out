# austin: 3.0.0
# interval: 10000
# mode: wall

P82848;T82848 21
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10058
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:37;/usr/lib/python3.8/threading.py:start:857;/usr/lib/python3.8/threading.py:wait:558;/usr/lib/python3.8/threading.py:wait:302 10058
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10083
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10083
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10088
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10088
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10088
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10088
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10087
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10087
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10060
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10060
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10085
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:30 10085
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10082
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10082
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10020
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10020
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10090
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10090
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10057
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:30 10057
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10058
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10058
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10087
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10087
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10087
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10087
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10083
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10083
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10060
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:30 10060
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:30 10057
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10057
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10083
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10083
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10094
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10094
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10058
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10058
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10058
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10058
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10057
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10057
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10081
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10081
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10087
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10087
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10087
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10087
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10087
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10087
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10081
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10081
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10088
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10088
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:30 10084
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10084
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10089
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10089
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10089
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10089
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10084
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:31 10084
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10088
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10088
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870 10090
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10090
P82848;T82858;/usr/lib/python3.8/threading.py:_bootstrap:890;/usr/lib/python3.8/threading.py:_bootstrap_inner:932;/usr/lib/python3.8/threading.py:run:870 10091
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10091
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38;/home/gabriele/Projects/austin/test/target34.py:keep_cpu_busy:32 10058
P82848;T82848;/home/gabriele/Projects/austin/test/target34.py:<module>:38 10092

# duration: 383010

